crates/client-api/src/routes/subscribe.rs

```diff
@@ -257,7 +262,11 @@ async fn ws_client_actor_inner(
 
             // If the module has exited, close the websocket.
             () = client.module.exited(), if !closed => {
-                if let Err(e) = ws.close(Some(CloseFrame { code: CloseCode::Away, reason: "module exited".into() })).await {
+                let close = also_poll(
```

Comment:
```suggestion
                // Send a close frame while continuing to poll the `handle_queue`,
                // to avoid deadlocks or delays due to enqueued futures holding resources.
                let close = also_poll(
```